File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] resArray = new int[2];
        HashMap<Integer,Integer> hm = new HashMap<Integer,Integer>();
        for(int i=0; i<nums.length; i++){
            if(hm.containsKey(nums[i])){
                resArray[0]=hm.get(nums[i]);
                resArray[1]=i;
                return resArray;
            }else{
                int res = target-nums[i];
                hm.put(res, i);
            }
        }
        return resArray;
    }
}

File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let res: number[] = [];
    let visited= new Map<number, number>(); // value required, index
    for (let i = 0; i < nums.length; i++) {
        const curr = nums[i];
        
        if(visited.has(curr)){
            res[0] = visited.get(curr);
            res[1] = i;
            return res;
        }
        else{
            visited.set(target-curr,i);
        }
    }

    return res;
};